elide-core/src/main/java/com/yahoo/elide/parsers/state/StartState.java

```diff
@@ -68,7 +70,7 @@ public void handle(StateContext state, RootCollectionRelationshipContext ctx) {
         String id = ctx.entity().id().getText();
         Class<?> entityClass = dictionary.getEntityClass(entityName);
         if (entityClass == null || !dictionary.isRoot(entityClass)) {
-            throw new InvalidCollectionException(entityName);
+            throw new ForbiddenEntityException();
```

Comment:
Should we be throwing ForbiddenAccessException here since we are in the process of fetching a relationship?

Reply:
Should we also rethrow the InvalidCollectionException below with a ForbiddenAccessException, for the same reason.

Reply:
Yes indeed, good catch.

Reply:
This still needs an update.

Reply:
Addressed.